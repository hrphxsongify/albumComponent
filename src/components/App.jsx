import React from 'react';
import $ from 'jquery';
import TrackContainer from './TrackContainer.jsx';
import AlbumInfo from './AlbumInfo.jsx';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentAlbumData: {
        name: '',
        year: 0,
        tracks: []
      }
    };
  }

  componentDidMount() {
    const window = document.URL;
    const url = window[window.length - 1];
    $.ajax({
      type: 'get',
      url: `http://localhost:3002/api/albums/${url}`,
      success: albumData => {
        this.setState({
          currentAlbumData: albumData
        });
      },
      dataType: 'json'
    });
  }

  render() {
    return (
      <div>
        <AlbumInfo albumInfo={this.state.currentAlbumData} />
        <TrackContainer
          tracks={this.state.currentAlbumData.tracks}
          artist={this.state.currentAlbumData.artist}
        />
      </div>
    );
  }
}

export default App;
