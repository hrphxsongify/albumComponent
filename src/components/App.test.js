import React from 'react';
import { shallow } from 'enzyme';

import App from './App.jsx';

describe('App', () => {
  it('stuff', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });
});
