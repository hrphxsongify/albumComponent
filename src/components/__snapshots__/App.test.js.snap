// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component render 1`] = `
<div>
  <AlbumInfo
    albumInfo={
      Object {
        "name": "",
        "tracks": Array [],
        "year": 0,
      }
    }
  />
  <TrackContainer
    tracks={Array []}
  />
</div>
`;
